channeldb/channel.go

```diff
@@ -3869,6 +3869,7 @@ func fetchChanInfo(chanBucket kvdb.RBucket, channel *OpenChannel) error {
 			initialRemoteBalanceType, &remoteBalance,
 		),
 		MakeScidRecord(realScidType, &channel.confirmedScid),
+		tlv.MakePrimitiveRecord(channelMemoType, &channel.Memo),
```

Comment:
I was able to track down the weird crash in the unit test to this. I think you need to use an extra variable here (e.g. `var memo []byte`, decode into that (`tlv.MakePrimitiveRecord(channelMemoType, &memo),` and then check if something was decoded:
```
if len(memo) > 0 {
    channel.Memo = memo
}
```

This seems to be a weird `nil` slice vs. empty slice thing in the deep comparison...

Reply:
Thanks for looking into that! I will try your suggestion :slightly_smiling_face: